modules/olm-disconnected-operatorhub-disconnected-mode.adoc

```diff
@@ -0,0 +1,171 @@
+// Module included in the following assemblies:
+//
+// * applications/operators/olm-disconnected.adoc
+
+[id="olm-disconnected-operatohub-disconnected-mode_{context}"]
+= Configuring OperatorHub in disconnected mode
+
+Cluster administrators can configure OLM and OperatorHub to use local content in
+a disconnected mode.
+
+.Prerequisites
+
+* Cluster administrator access to a connected or disconnected {product-title} cluster and its internal registry.
+* Separate workstation with full Internet access.
+
+.Procedure
+
+. **Disable the default OperatorSources.**
+
+.. Edit the `OperatorHub` configuration resource:
++
+----
+$ oc edit OperatorHub cluster
+----
+
+.. Add `disableAllDefaultSources: true` to the spec:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: OperatorHub
+spec:
+  disableAllDefaultSources: true
+----
++
+This disables the default OperatorSources that are configured by default during
+a {product-title} installation.
+
+. **Retrieve package lists.**
++
+To get the list of packages that are available for the default OperatorSources,
+run the following `curl` commands from a workstation with Internet access:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages?namespace=redhat-operators > packages.txt
+$ curl https://quay.io/cnr/api/v1/packages?namespace=community-operators >> packages.txt
+$ curl https://quay.io/cnr/api/v1/packages?namespace=certified-operators >> packages.txt
+----
++
+Each package in the new `packages.txt` is an Operator that you could add to your
+disconnected catalog. From this list, you could either pull every Operator or a
+subset that you would like to expose to users.
+
+. **Pull Operator content.**
++
+For a given Operator in the package list, you must pull its content:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages/<namespace>/<operator_name>/<release>
+----
++
+This example uses the etcd Operator:
+
+.. Retrieve the digest:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages/community-operators/etcd/0.0.12
+----
+
+.. From that JSON, take the digest and use it to pull the gzipped archive:
++
+----
+$ curl -XGET https://quay.io/cnr/api/v1/packages/community-operators/etcd/blobs/sha256/8108475ee5e83a0187d6d0a729451ef1ce6d34c44a868a200151c36f3232822b -o etcd.tar.gz
+----
+
+.. To pull the information out, you must untar the archive into a directory with
+all the other Operators that you want. For example, to untar to an existing
+directory called `manifests/etcd/`:
++
+----
+$ mkdir manifests/etcd/ <1>
+$ tar -xf etcd.tar.gz -C manifests/etcd/
+----
+<1> Create different subdirectories for each extracted archive so that files are not
+overwritten by subsequent extractions for other Operators.
+
+.. In this directory, you must now handle two different types of bundles:
+
+... If you see a single file called `bundle.yaml`, you must break the content under
```

Comment:
I'm really concerned about letting the user do this step, but I know that are improvements being implemented. Is it possible to add the bundle.yaml in this doc used for this example? That would help the user to associate with their use cases.

Reply:
@bandrade I've added step 4 about how to break up the file to http://file.rdu.redhat.com/~adellape/083019/olm_disconnected/applications/operators/olm-disconnected.html.

Reply:
Thanks @adellape, lgmt.